### ISSUE TITLE
Expected results after training on the joint set of CrowdHuman and MOT17

### ISSUE THREAD

**Issue by FengLi-ust** [timmeinhardt/trackformer#46]
## Expected results after training on the joint set of CrowdHuman and MOT17
Hey, thank you for you excellent work!

I train TrackFormer on your default setting (load from pretrained CrowdHuaman checkpoint) on the joint set of CrowdHuman and MOT17, but get result on MOT17 73.3 MOTA (I think it is validated on your default dataset mot17_train_cross_val_frame_0_5_to_1_0_coco). 

Is the result corresponds to the training set result (74.2 MOTA provided in ReadMe) or the cross-validate result (71.3 MOTA provided in the paper)?

Thank you in advance.

**Comment by timmeinhardt** [comment]
Hello, this all depends on the training and validation set. The 74.2 MOTA is trained on `CrowdHuman + mot17_train_coco`  and evaluated on the full training set `mot17_train_coco`. If you validate on `mot17_train_cross_val_frame_0_5_to_1_0_coco ` you need to set the training set to `CrowdHuman + mot17_train_cross_val_frame_0_0_to_0_5_coco `.  With which setup did you obtain 73.3 MOTA?

**Comment by FengLi-ust** [comment]
Thanks for your reply. I train with your default setting on `CrowdHuman + mot17_train_coco` and do not change the validation set or other code. 
`
python src/train.py with \
    mot17_crowdhuman \
    deformable \
    multi_frame \
    tracking \
    output_dir=models/mot17_crowdhuman_deformable_multi_frame \
`
I check your code and I think the default validation set is `mot17_train_cross_val_frame_0_5_to_1_0_coco`. Can you confirm the validation set?

**Comment by timmeinhardt** [comment]
The default validation set is indeed `mot17_train_cross_val_frame_0_5_to_1_0_coco `. But if you train on `CrowdHuman + mot17_train_coco ` you are training on the full MOT17 dataset. In that case, there is nothing to validate on that`s why we just left the ablation validation set. If you want to reproduce the ablation results you need to train on `(CrowdHuman) + mot17_train_cross_val_frame_0_0_to_0_5_coco `. Please note the difference `0_0_to_0_5` vs. `0_5_to_1_0`.

**Comment by timmeinhardt** [comment]
Yes, that is correct. For all ablations, you need to train on `mot17_train_cross_val_frame_0_0_to_0_5_coco` and validate on `mot17_train_cross_val_frame_0_5_to_1_0_coco `. And for the final model you need to add `CrowdHuman` to the training.

We experimented with `prev_prev_image` but it did not produce better results. 

A unified encoder might improve results but we kept the encoder separate mainly for computational reasons. Computing self-attention on multi-scale features of two frames at once is very expensive.

**Comment by FengLi-ust** [comment]
Thank you for your explanation. Issue closed.

**Comment by quxu91** [comment]
@FengLi-ust  I met the same problem. After reading the above conversation, I would like to know how to set the parameters if I want to reproduce the results of 74.2 MOTA（provided in ReadMe）?
